client/src/components/Ticketing/ticketingmanager/AdminPurchase/ReaderPurchase.tsx

```diff
@@ -0,0 +1,239 @@
+import React, {useEffect, useState} from 'react';
+import {
+  Button,
+} from '@mui/material';
+import PopUp from '../../PopUp';
+import {useNavigate, useParams, useLocation} from 'react-router-dom';
+import useWebSocket from 'react-use-websocket';
+
+import {useAppDispatch, useAppSelector} from '../../app/hooks';
+
+import {useFetchToken} from '../Event/components/ShowingUtils';
+
+import {
+  removeAllTicketsFromCart,
+  selectDiscount,
+} from '../ticketingSlice';
+
+import {loadStripe} from '@stripe/stripe-js';
+
+const pk = `${process.env.REACT_APP_PUBLIC_STRIPE_KEY}`;
+const stripePromise = loadStripe(pk);
+
+const ReaderPurchase = () => {
+  const {token} = useFetchToken();
+  const [status, setStatus] = useState('Awaiting Response...');
+
+  const location = useLocation();
+  const navigate = useNavigate();
+  const cartItems = location.state?.cartItems || [];
+  const clientSecret = location.state?.clientSecret || '';
+  const readerID = location.state?.readerID || '';
+  const discount = useAppSelector(selectDiscount);
+  const [openDialog, setDialog] = useState(false);
+  const [errMsg, setErrMsg] = useState('');
+
+  const event = 'reader';
+  const paymentIntentID = useParams().id;
+
+  // const socketURL = process.env.WEBSOCKET_URL; use this once the env variable works
+  const socketURL = process.env.REACT_APP_ROOT_URL.replace('https', 'wss') + '/wss';
+
+  const {getWebSocket} = useWebSocket(socketURL, {
+    shouldReconnect: () => true,
+    onMessage: (event) => {
+      const ws = getWebSocket();
+      const data = JSON.parse(event.data);
+      if (data.messageType === 'reader' &&
+          data.paymentIntent === paymentIntentID ||
+          data.paymentIntent === readerID) { // in the case of terminal event, paymentIntent is a readerID
+        console.log(data);
+        setStatus(data.eventType);
+        if (data.eventType === 'payment_intent.succeeded') {
+          ws.close();
+          navigate(`/success`);
+        } else if (data.eventType === 'payment_intent.requires_action') { // not sure if this needs to be handelled but I think this is how its done
+          stripePromise.then((stripe) => {
+            if (!stripe) return;
+            stripe.confirmCardPayment(clientSecret).then((result) => {
+              console.log(result); // should handle failure/success here
+            });
+          });
+        } else if (data.eventType === 'terminal.reader.action_failed') {
+          ws.close();
+          setErrMsg(data.errMsg + ' Order canceled. Please Try again.');
+          setDialog(true);
+        }
+      }
+    },
+  });
+
+  useEffect(() => {
+    const processPayment = async () => {
+      try {
+        console.log({cartItems, paymentIntentID, clientSecret, readerID, discount});
```

Comment:
These `console.log()`s are fine for debugging locally but should be removed before this is merged and deployed

Reply:
*the console logs in the server are fine but the one's in the client are user facing

Reply:
Removed console.logs in client